src/com/twitter/intellij/pants/service/project/PythonVenvBuilder.java

```diff
@@ -0,0 +1,77 @@
+// Copyright 2020 Pants project contributors (see CONTRIBUTORS.md).
+// Licensed under the Apache License, Version 2.0 (see LICENSE).
+
+package com.twitter.intellij.pants.service.project;
+
+import com.intellij.execution.ExecutionException;
+import com.intellij.execution.configurations.GeneralCommandLine;
+import com.intellij.execution.process.ProcessAdapter;
+import com.intellij.execution.process.ProcessOutput;
+import com.intellij.openapi.diagnostic.Logger;
+import com.twitter.intellij.pants.service.project.model.PythonInterpreterInfo;
+import com.twitter.intellij.pants.util.PantsUtil;
+import org.jetbrains.annotations.NotNull;
+
+import java.nio.file.Path;
+import java.util.Collection;
+import java.util.Optional;
+
+public class PythonVenvBuilder {
+
+  protected static final Logger LOG = Logger.getInstance(PythonVenvBuilder.class);
+
+  private final String projectPath;
+  private final ProcessAdapter processAdapter;
+
+  private PythonVenvBuilder(String projectPath, ProcessAdapter processAdapter) {
+    this.projectPath = projectPath;
+    this.processAdapter = processAdapter;
+  }
+
+  public static Optional<PythonVenvBuilder> forProjectPath(String projectPath, ProcessAdapter processAdapter){
+    Collection<String> allTargets = PantsUtil.listMatchingTargets(projectPath, "entsec/venv_builder:venv_builder");
```

Comment:
Shouldn't the target name be configurable?

Reply:
The target name is always the same. If your project doesn't have this exact target, you probably can't use the venv_builder tool anyway.  Therefore, I think it is permissible to hardcode this target name.

However, the tool itself doesn't seem to contain any special logic. In the future, we might want to get that tool released to the public, or perhaps create a similar one.